modules/extensions-spring-support/src/test/java/org/mule/module/extension/internal/connector/PetStoreRetryPolicyConnectionTestCase.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.connector;
+
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+
+import org.mule.api.MessagingException;
+import org.mule.api.connection.ConnectionException;
+import org.mule.api.connection.ConnectionExceptionCode;
+import org.mule.api.connection.ConnectionHandlingStrategy;
+import org.mule.api.connection.ConnectionHandlingStrategyFactory;
+import org.mule.api.connection.ConnectionValidationResult;
+import org.mule.extension.annotation.api.Alias;
+import org.mule.extension.annotation.api.Extension;
+import org.mule.extension.annotation.api.Operation;
+import org.mule.extension.annotation.api.Operations;
+import org.mule.extension.annotation.api.capability.Xml;
+import org.mule.extension.annotation.api.connector.Providers;
+import org.mule.extension.annotation.api.param.Connection;
+import org.mule.functional.junit4.ExtensionFunctionalTestCase;
+import org.mule.module.extension.internal.runtime.connector.petstore.PetStoreClient;
+import org.mule.module.extension.internal.runtime.connector.petstore.PetStoreConnectionProvider;
+import org.mule.module.extension.internal.runtime.connector.petstore.PetStoreConnector;
+import org.mule.module.extension.internal.runtime.connector.petstore.PetStoreOperations;
+import org.mule.module.extension.internal.runtime.connector.petstore.PetStorePoolingProfile;
+import org.mule.retry.RetryPolicyExhaustedException;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class PetStoreRetryPolicyConnectionTestCase extends ExtensionFunctionalTestCase
+{
+
+    public static final String CONNECTION_FAIL = "Connection fail";
+    public static final String CONNECTION_FAIL_DOT = "Connection fail.";
```

Comment:
Because a dot is added to the original message, so when asserting the error message, that makes they differ.